src/test/rbd_mirror/test_mock_LeaderWatcher.cc

```diff
@@ -53,49 +61,166 @@ struct ManagedLock<MockTestImageCtx> {
               const std::string& oid, librbd::Watcher *watcher,
               managed_lock::Mode  mode, bool blacklist_on_break_lock,
               uint32_t blacklist_expire_seconds)
-    : m_lock("ManagedLock::m_lock") {
+    : m_work_queue(work_queue), m_lock("ManagedLock::m_lock") {
+    MockManagedLock::get_instance().construct();
+  }
+
+  virtual ~ManagedLock() {
```

Comment:
@dillaman ManagedLock is a base (not derived) class. I think "virtual" is correct in this case?